mso/resource_mso_schema.go

```diff
@@ -375,7 +375,11 @@ func resourceMSOSchemaUpdate(d *schema.ResourceData, m interface{}) error {
 				for _, valueMapNew := range getDifferenceNew {
 					valueNew := valueMapNew.(map[string]interface{})
 					if valueOld["name"] == valueNew["name"] && valueOld["template_type"] != valueNew["template_type"] {
-						return fmt.Errorf("Template type cannot be changed.")
+						// Conditional to check if template type configured is aci_multisite and template type not provided
+						// This conditional is added to handle default behaviour when template type is not provided
+						if valueNew["template_type"] != "" && valueOld["template_type"] == "aci_multisite" {
+							return fmt.Errorf("Template type cannot be changed.")
```

Comment:
Are we allowing the user to change the template_type for other scenarios apart from the above condition?

Reply:
adjusted by leveraging default since that wil remove the need of complex conditionals which in the end amount to same thing